### DISCUSSION TITLE
v0.6.10

### DISCUSSION THREAD

**Discussion by takatost** [langgenius/dify#4934] [Releases]
## ✨ What’s New in v0.6.10? ✨

This update primarily focuses on enhancing system stability and improving user experience. Key updates include:

### 🚀 New Features

- **Workflow Variable Aggregator**: Now supports grouping by @zxhlyh in #4811.
- **Document Rename Support**: You can now rename documents in datasets by @iamjoel in #4732 and @JohnJyong in #4915.
- **Instruction in Classifier Node**: Added support for using variables in instructions by @iamjoel in #4710 and @JohnJyong in #4913.
- **Tool Tags**: You can now define tags in tool YAML files by @Yeuoly in #4763.
- **NVIDIA NIM**: Added support for NVIDIA NIM by @joshua20231026 in #4882.
- **Anthropic Claude3 Models**: Added to Google Cloud Vertex AI by @hellof20 in #4870.
- **Baichuan Model Support**: Added support for Baichuan3 Turbo, Baichuan3 Turbo 128k, and Baichuan4 by @xielong in #4762.
- **Ernie Model Update**: Updated the Ernie model by @Weaxs in #4756.
- **Vanna.AI Tool**: Added Vanna.AI as a built-in tool by @luxmus in #4878.
- **SearchApi Tools**: Added new SearchApi tools by @SebastjanPrachovskij in #4648.
- **Opportunistic TLS for SMTP**: Added a flag for opportunistic TLS in SMTP by @chazzhou in #4794.
- **Ollama Keep Alive**: Added a keep-alive parameter for Ollama by @Yash-1511 in #4655.
- **Model Load Balancing**: Added support for load balancing backend models by @takatost in #4927 and @nite-knite in #4926.

### ⚙️ Enhancements

- **SD-WebUI API Update**: Updated API parameters to v1.9.3 by @lingfengchencn in #4798.
- **Pandas Upgrade**: Bumped pandas from 1.x to 2.x by @bowenliang123 in #4820.
- **Question Classify Node**: Now supports using variables in instructions by @iamjoel in #4710.

For additional enhancements and updates, refer to the What's Changed section below.

### Docker compose deployments:

1. Get the latest code from the main branch:

   ```Shell
   git checkout main
   git pull origin main
   ```

2. Go to the next step and update to the latest image:

   ```Shell
   cd docker
   docker compose up -d
   ```

### Source Code deployments:

1. Stop API server, Worker and Web frontend Server.

2. Get the latest code from the main branch:

   ```Shell
   git checkout main
   git pull origin main
   ```

3. Update Python dependencies:

   ```Shell
   cd api
   pip install -r requirements.txt
   ```

4. Then, let's run the migration script:

   ```Shell
   flask db upgrade
   ```

5. Finally, run API server, Worker and Web frontend Server again.

## What's Changed
* feat: add América/São Paulo tz by @crazywoola in https://github.com/langgenius/dify/pull/4701
* Add workflow translations for ja-JP by @Hirosa0 in https://github.com/langgenius/dify/pull/4698
* fix: key-value-table styles by @HiChen404 in https://github.com/langgenius/dify/pull/4678
* fix: Input fields in the model provider's settings modal do not switch sequence via keyboard navigation (Tab key)  by @doufa in https://github.com/langgenius/dify/pull/4662
* fix: the new node name is generated based on the original node when duplicating by @hjlarry in https://github.com/langgenius/dify/pull/4675
* fix: workflow as tool garbled by @Yeuoly in https://github.com/langgenius/dify/pull/4707
* style: update VarPanel to use whitespace-pre-wrap for value display by @xuzuodong in https://github.com/langgenius/dify/pull/4684
* add error msg for hit test by @JohnJyong in https://github.com/langgenius/dify/pull/4704
* fix: retrieval value greater more than 1 caused ui problem by @iamjoel in https://github.com/langgenius/dify/pull/4718
* fix: Correct context size for banchuan2-53b and banchuan2-turbo by @xielong in https://github.com/langgenius/dify/pull/4721
* Show tool i18n name on chat pannel by @hibernate2011 in https://github.com/langgenius/dify/pull/4724
* fix: workflow run sequence number slow sql by @takatost in https://github.com/langgenius/dify/pull/4737
* fix: optimize sticky header styles z-index in tools - ProviderList component by @chazzhou in https://github.com/langgenius/dify/pull/4746
* fix: Corrected schema link in model_runtime's README.md by @xielong in https://github.com/langgenius/dify/pull/4757
* style: the 'all' of add tool panel should contain workflow tools by @hjlarry in https://github.com/langgenius/dify/pull/4755
* Fix/4742 ollama num gpu option not consistent with allowed values by @crazywoola in https://github.com/langgenius/dify/pull/4751
* Add WORKFLOW_CALL_MAX_DEPTH env var. by @majian159 in https://github.com/langgenius/dify/pull/4713
* style: fix annotation panel display misalignment by @hjlarry in https://github.com/langgenius/dify/pull/4750
* fix: confusing chart description by @Yeuoly in https://github.com/langgenius/dify/pull/4760
* feat: support baichuan3 turbo, baichuan3 turbo 128k, and baichuan4 by @xielong in https://github.com/langgenius/dify/pull/4762
* fix: incorrect workflow max call depth by @Yeuoly in https://github.com/langgenius/dify/pull/4759
* feat: update ernie model by @Weaxs in https://github.com/langgenius/dify/pull/4756
* feat: support define tags in tool yaml by @Yeuoly in https://github.com/langgenius/dify/pull/4763
* fix organize agent's history messages without recalculating tokens by @zeroameli in https://github.com/langgenius/dify/pull/4324
* feat: Add logging warning when MAIL_TYPE is not set by @Kota-Yamaguchi in https://github.com/langgenius/dify/pull/4771
* fix: in tool and http node of iteration can not show item var correctly by @iamjoel in https://github.com/langgenius/dify/pull/4791
* chore: improve node user experience by @iamjoel in https://github.com/langgenius/dify/pull/4792
* Add SearchApi tools by @SebastjanPrachovskij in https://github.com/langgenius/dify/pull/4648
* chore: node help link by @zxhlyh in https://github.com/langgenius/dify/pull/4795
* fix: some filed in model param selector has no left spacing  by @iamjoel in https://github.com/langgenius/dify/pull/4803
* feat: show more usage info in billing page by @iamjoel in https://github.com/langgenius/dify/pull/4808
* add: i18n: update korean by @qkdxorjs1002 in https://github.com/langgenius/dify/pull/4813
* feat: workflow variable aggregator support group by @zxhlyh in https://github.com/langgenius/dify/pull/4811
* feat: opportunistic tls flag for smtp by @chazzhou in https://github.com/langgenius/dify/pull/4794
* fix: remove redundant props by @lesenelir in https://github.com/langgenius/dify/pull/4787
* update sd-webui api parameters to v1.9.3 by @lingfengchencn in https://github.com/langgenius/dify/pull/4798
* fix: revision styles for workflow by @lawvs in https://github.com/langgenius/dify/pull/4087
* fix: app_count of dataset is error when apps was deleted by @DomKing in https://github.com/langgenius/dify/pull/4810
* Feat/i18n workflow by @crazywoola in https://github.com/langgenius/dify/pull/4819
* chore: Enable case-insensitive search for large models by @xielong in https://github.com/langgenius/dify/pull/4817
* fix: workflow app run by @zxhlyh in https://github.com/langgenius/dify/pull/4831
* add: ollama keep alive parameter added. issue #4024 by @Yash-1511 in https://github.com/langgenius/dify/pull/4655
* fix 'NoneType' and new ContentType supported. by @lichengwu in https://github.com/langgenius/dify/pull/4818
* chore: cleanup tools, remove useless code by @Yeuoly in https://github.com/langgenius/dify/pull/4833
* chore: increase workflow max steps to 500 by @Yeuoly in https://github.com/langgenius/dify/pull/4835
* fix: status query not stop when leaving document embedding detail page by @doufa in https://github.com/langgenius/dify/pull/4754
* test: triggering tests on changes and allow cancelling in-progress CI test jobs by @bowenliang123 in https://github.com/langgenius/dify/pull/4743
* test: CI test for db migration scripts on changes by @bowenliang123 in https://github.com/langgenius/dify/pull/4739
* chore: skip explicit installing jinja2 as testing dependency by @bowenliang123 in https://github.com/langgenius/dify/pull/4845
* chore: fix invalid escape sequences by applying W605 rule by @bowenliang123 in https://github.com/langgenius/dify/pull/4851
* Bugfix: Vertex AI vision model not support image by @hellof20 in https://github.com/langgenius/dify/pull/4853
* chore: upgrade sandbox by @Yeuoly in https://github.com/langgenius/dify/pull/4839
* fix: Incorrect argument types in createChatMessage by @dar0xt in https://github.com/langgenius/dify/pull/4861
* Fixed workflow tts feature audition by @charli117 in https://github.com/langgenius/dify/pull/4867
* fix: update npm version to fix Incorrect argument types in createChatMessage by @iamjoel in https://github.com/langgenius/dify/pull/4865
* chore: add issue link tempate for IDEA by @bowenliang123 in https://github.com/langgenius/dify/pull/4866
* add-nvidia-mim by @joshua20231026 in https://github.com/langgenius/dify/pull/4882
* improve: generalize tool parameter converter by @bowenliang123 in https://github.com/langgenius/dify/pull/4786
* chore: optimize nvidia nim credential schema and info by @takatost in https://github.com/langgenius/dify/pull/4898
* feat: added Anthropic Claude3 models to Google Cloud Vertex AI by @hellof20 in https://github.com/langgenius/dify/pull/4870
* chore: remove unused code and class in text splitter by @bowenliang123 in https://github.com/langgenius/dify/pull/4864
* chore(deps): bump azure-storage-blob from 12.9.0 to 12.13.0 in /api by @dependabot in https://github.com/langgenius/dify/pull/4695
* chore: separate style checks into multiple jobs triggering on file changes by @bowenliang123 in https://github.com/langgenius/dify/pull/4876
* chore: update page.tsx by @eltociear in https://github.com/langgenius/dify/pull/4897
* fix azure blob token expire by @JohnJyong in https://github.com/langgenius/dify/pull/4911
* chore: modify tools/JinaReader label to Jina by @doufa in https://github.com/langgenius/dify/pull/4908
* dep: bump pandas from 1.x to 2.x by @bowenliang123 in https://github.com/langgenius/dify/pull/4820
* Fixed a bug where any content in the 'fetch' was converted to True by @huangbaichao in https://github.com/langgenius/dify/pull/4400
* feat: question classify node support use var in instruction by @iamjoel in https://github.com/langgenius/dify/pull/4710
* chore: refactor the serpapi's google search tool by @hjlarry in https://github.com/langgenius/dify/pull/4834
* feat: Add Vanna.AI as a builtin tool by @luxmus in https://github.com/langgenius/dify/pull/4878
* Fix/azure blob token expire by @JohnJyong in https://github.com/langgenius/dify/pull/4914
* feat: document support rename in in dataset by @iamjoel in https://github.com/langgenius/dify/pull/4732
* chore:  make the error msg more clear when validate app token by @hjlarry in https://github.com/langgenius/dify/pull/4919
* fix: router replace in Explore page by @ZuzooVn in https://github.com/langgenius/dify/pull/4918
* modify rerank and splitter code directory by @JohnJyong in https://github.com/langgenius/dify/pull/4924
* support rename document by @JohnJyong in https://github.com/langgenius/dify/pull/4915
* support instruction in classifier node by @JohnJyong in https://github.com/langgenius/dify/pull/4913
* fix: incorrect parameters transforming while validating by @Yeuoly in https://github.com/langgenius/dify/pull/4928
* feat: backend model load balancing support by @takatost in https://github.com/langgenius/dify/pull/4927
* feat: model load balancing  by @nite-knite in https://github.com/langgenius/dify/pull/4926
* fix: vanna CVE-2024-5565 by disable visualize of ask func by @takatost in https://github.com/langgenius/dify/pull/4930
* version to 0.6.10 by @takatost in https://github.com/langgenius/dify/pull/4929
* fix: question classifier instruction npe by @takatost in https://github.com/langgenius/dify/pull/4931
* fix: bedrock get_num_tokens prompt_messages parameter name err by @takatost in https://github.com/langgenius/dify/pull/4932
* pref: optimize feature model_load_balancing_enabled value fetch speed… by @takatost in https://github.com/langgenius/dify/pull/4933

## New Contributors
* @Hirosa0 made their first contribution in https://github.com/langgenius/dify/pull/4698
* @doufa made their first contribution in https://github.com/langgenius/dify/pull/4662
* @xuzuodong made their first contribution in https://github.com/langgenius/dify/pull/4684
* @xielong made their first contribution in https://github.com/langgenius/dify/pull/4721
* @zeroameli made their first contribution in https://github.com/langgenius/dify/pull/4324
* @Kota-Yamaguchi made their first contribution in https://github.com/langgenius/dify/pull/4771
* @SebastjanPrachovskij made their first contribution in https://github.com/langgenius/dify/pull/4648
* @lesenelir made their first contribution in https://github.com/langgenius/dify/pull/4787
* @DomKing made their first contribution in https://github.com/langgenius/dify/pull/4810
* @lichengwu made their first contribution in https://github.com/langgenius/dify/pull/4818
* @hellof20 made their first contribution in https://github.com/langgenius/dify/pull/4853
* @dar0xt made their first contribution in https://github.com/langgenius/dify/pull/4861
* @luxmus made their first contribution in https://github.com/langgenius/dify/pull/4878

**Full Changelog**: https://github.com/langgenius/dify/compare/0.6.9...0.6.10

<hr /><em>This discussion was created from the release <a href='https://github.com/langgenius/dify/releases/tag/0.6.10'>v0.6.10</a>.</em>

**Comment by brightwang-igg** [comment]
npm run build error

Import trace for requested module:
./app/components/tools/provider/detail.tsx
./app/components/tools/provider-list.tsx
./app/(commonLayout)/tools/page.tsx

./app/components/tools/provider/detail.tsx
Attempted import error: 'ConfigurateMethodEnum' is not exported from '@/app/components/header/account-setting/model-provider-page/declarations' (imported as 'ConfigurateMethodEnum').

Import trace for requested module:
./app/components/tools/provider/detail.tsx
./app/components/tools/provider-list.tsx
./app/(commonLayout)/tools/page.tsx

**Reply by arcelan** [reply]
seems to be a bug, "ConfigurateMethodEnum" should be corrected as "ConfigurationMethodEnum"

**Reply by nite-knite** [reply]
This issue has been fixed in #4944

**Comment by arcelan** [comment]
just want to know if threre any docs about "model load balancing"?
ex. some screenshots?

**Comment by yanxiaochen** [comment]
有人试过可以添加最新开源的GML4-9B-12BK嘛
